### PR TITLE
Log errors in JSON reader to accountant

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: libvast/vast/format/json.hpp [hunk 1]
```diff
@@ -115,6 +117,8 @@ class reader final : public multi_layout_reader {
   std::unique_ptr<detail::line_range> lines_;
   caf::optional<size_t> proto_field_;
   std::vector<size_t> port_fields_;
+  mutable size_t invalid_line_count_ = 0;
+  mutable size_t unknown_layout_count_ = 0;
```
Comment (mavam) [inline libvast/vast/format/json.hpp:121]: We usually use `num_invalid_lines_` for counter-like variables. (This should go in the style guide, actually.)

File: libvast/vast/format/json.hpp [hunk 2]
```diff
@@ -150,6 +154,19 @@ const char* reader<Selector>::name() const {
...
+template <class Selector>
+vast::system::report reader<Selector>::status() const {
+  using namespace std::string_literals;
+  uint64_t invalid_line = invalid_line_count_;
+  uint64_t unknown_layout = unknown_layout_count_;
+  invalid_line_count_ = 0;
+  unknown_layout_count_ = 0;
+  return {
+    {name() + ".invalid-line"s, invalid_line},
+    {name() + ".unknown_layout"s, unknown_layout},
```
Comment (mavam) [inline libvast/vast/format/json.hpp:166]: Shouldn't these be pluralized too?

Comment (dominiklohmann) [reply libvast/vast/format/json.hpp:166]: They aren't currently to stay in line with the PCAP reader.

File: libvast/vast/format/json.hpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/vast/format/json.hpp:121]: Alright, will rename.

File: libvast/vast/format/json.hpp (hunk 2 above)
Comment (mavam) [reply libvast/vast/format/json.hpp:166]: Okay.

**Pull request merged**